tests/integration/test_source_git.py

```diff
@@ -105,7 +109,12 @@ def test_basic_local_update_patch_content(
     source_file.write_text(" And I am sad.")
     git_add_and_commit(directory=sourcegit, message="make a file sad")
 
-    api_instance_source_git.sync_release("master", "0.1.0", upstream_ref="0.1.0")
+    api_instance_source_git.sync_release(
+        dist_git_branch="master",
+        version="0.1.0",
+        use_local_content=True,
+        upstream_ref="0.1.0",
+    )
```

Comment:
> I added a commit with changes so that the tests are passing, please check, for some test cases, I added the use_local_content=True in sync_release since the tag was added in the fixture and then in the test cases there where added other commits, therefore when the tag was checked out in sync_release, it was not in the expected state.

I am not sure if I got that correctly.

- What is expected state? Do we care that we don't have content from the next commit when syncing the release?
- Was the change in fixture needed?
- I am not against that change but want to be sure we are not hiding any real problem with this.

Reply:
> What is expected state? Do we care that we don't have content from the next commit when syncing the release?

I didn't write the original test for it, but if  I understand it correctly, we want to [check out](https://github.com/packit/packit/blob/f2899ec65ef943ba3d9496ba7b5074a2defd829b/packit/api.py#L188) the tag with these commits so that we can work with the state including them.

> Was the change in fixture needed?

I assume that we want to have `source_git` directory present in the repo when we check out the `0.1.0` tag, so then yes. (Before this change the `0.1.0` tag was created and the `source_git` repo added just after that.)

Reply:
The source-git should be:

- upstream history + upstream-only content with tagged version
- on top of it, we are adding distribution-specific data/commits (=the content of `SOURCEGIT_SOURCEGIT`)

So, I would probably switch back the order in the fixture.
- If it works, it's nice.
- If not, let's see if it is a real problem or not.

Reply:
Okay, thanks for the explanation. But before switching the order in the fixture, tests [failed](https://github.com/packit/packit/pull/961#issuecomment-693220143), so then this is probably a real problem, isn't it?

Reply:
That's a good point! And I am really not sure what should be the best approach in that case.

I would use `--local-content` in the tests for now and discuss the propose-update for source-git repositories in another issue or architecture meeting.

Reply:
I'll write that down in another issue.

Can you please only revert that fixture change? And sorry for delaying your PR...;) Thanks for the fix!

Reply:
Here is an issue: https://github.com/packit/packit/issues/975